README.md

```diff
@@ -18,29 +18,131 @@ in the Gerrit review.
 Xlint parses the changes in a patchset, and runs each change through
 a validator. If the validator detects any issues, a [Gergich](https://rubygems.org/gems/gergich)
 comment is created. After all the changes have been checked, Xlint
-publishes the Gergich comments to Gerrit.
+publishes the Gergich comments to Gerrit. If the CI environment has a varible for a Gerrit review label, Xlint will also post a reply to the label.
 
 ## Limitations
 
 Xlint currently only detects changes to the deployment target within
 Xcode .pbxproj files.
 
-## Installation
+## Installation and Usage
 
 [Gergich][gergich] and Gerrit must be configured as defined in the Gergich gem. If
 [Gergich][gergich] works, then all you need to do is `gem install xlint` and Xlint
 is ready for linting.
 
 [gergich]: https://github.com/instructure/gergich
 
-## Usage
+### Setup Gerrit and Gergich Environment Variables on Jenkins
+* Install EnvInject Jenkins Plugin (Manage Jenkins > Manage Plugins > Available Filter: EnvInject)
+Note: The Jenkins version on Cloudbees does not currently work with EnvInject.
+* Add the following Global property (Jenkins Configuration > Global Properties)
 
-Xlint requires patchset changes be saved to a file and the filepath
-passed to Xlint as a command-line argument.
+> GERGICH_KEY
+```
+Name: GERGICH_KEY
+Value: <access_key_for_gergich_user_on_gerrit>
+```
+
+* Check "This project is parameterized" (Job Configuration > General)
+* Add the following String Parameters to (Job Configuration > General
+
+> GERRIT_REFSPEC
+```
+Name: GERRIT_REFSPEC
+Default Value: HEAD:refs/for
+```
+
+> GERRIT_BRANCH
+```
+Name: GERRIT_BRANCH
+Default Value: develop 
+```
 
-## Example
+> GERGICH_REVIEW_LABEL
+```
+Name: GERGICH_REVIEW_LABEL
+Default Value: Lint-Review
+```
 
+* Check "Inject environment variables to the build process" (Job Configuration > Build Environment)
+
+### Setup Lint-Review on Gerrit
+In the meta/config branch of your Gerrit project, you need to add "Linter Bots" to your groups file, create the Lint-Review label, and grant access to bots in the project.config file.
+
+> groups
+```
+# UUID                                Group Name
+#
+<uuid_for_developers_group>           Developers
+<uuid_for_robots_group>               Robots
+<uuid_for_administrators_group>       Administrators
+<uuid_for_linter_bots_group>          Linter Bots
 ```
+
+> project.config
+```
+[access "refs/*"]
+    owner = group Administrators
+    owner = group Developers
+    read = group Robots
+[label "Lint-Review"]
+    function = AnyWithBlock
+    abbreviation = L
+    value = -2 Error
+    value = -1 Warning
+    value =  0 No score
+    value = +1 Verified
+    defaultValue = 0
+[access "refs/heads/*"]
+    label-Lint-Review = -2..1 group Linter Bots
+```
+
+* How do project labels work? Please read https://gerrit.baserock.org/Documentation/config-labels.html
```

Comment:
link should be https://gerrit-documentation.storage.googleapis.com/Documentation/2.12.2/config-labels.html